Source/items.cpp

```diff
@@ -1806,35 +1807,65 @@ void DrawUniqueInfoWindow(const Surface &out)
 void PrintItemMisc(const Item &item)
 {
 	if (item._iMiscId == IMISC_SCROLL) {
-		AddPanelString(_("Right-click to read"));
+		if (ControlMode == ControlTypes::KeyboardAndMouse) {
+			AddPanelString(_("Right-click to read"));
+		} else {
+			if (item.IsScrollOf(SPL_TELEPORT) || item.IsScrollOf(SPL_TOWN)) {
+				AddPanelString(_("Select from spell book, then"));
+				AddPanelString(_("cast spell to read"));
+			} else {
+				AddPanelString(_("Activate to read"));
+			}
+		}
 	}
 	if (item._iMiscId == IMISC_SCROLLT) {
-		AddPanelString(_("Right-click to read, then"));
-		AddPanelString(_("left-click to target"));
-	}
-	if (item._iMiscId >= IMISC_USEFIRST && item._iMiscId <= IMISC_USELAST) {
-		PrintItemOil(item._iMiscId);
-		AddPanelString(_("Right-click to use"));
-	}
-	if (item._iMiscId > IMISC_OILFIRST && item._iMiscId < IMISC_OILLAST) {
-		PrintItemOil(item._iMiscId);
-		AddPanelString(_("Right click to use"));
+		if (ControlMode == ControlTypes::KeyboardAndMouse) {
+			AddPanelString(_("Right-click to read, then"));
+			AddPanelString(_("left-click to target"));
+		} else {
+			if (item.IsScrollOf(SPL_FLAME)
+			    || item.IsScrollOf(SPL_LIGHTNING)
+			    || item.IsScrollOf(SPL_GUARDIAN)
+			    || item.IsScrollOf(SPL_STONE)
+			    || item.IsScrollOf(SPL_FIREWALL)
+			    || item.IsScrollOf(SPL_FIREBALL)) {

```

Comment:
This list is missing Flame Wave.

```suggestion
			if (item.IsScrollOf(SPL_FLAME)
			    || item.IsScrollOf(SPL_WAVE)
			    || item.IsScrollOf(SPL_LIGHTNING)
			    || item.IsScrollOf(SPL_GUARDIAN)
			    || item.IsScrollOf(SPL_STONE)
			    || item.IsScrollOf(SPL_FIREWALL)
			    || item.IsScrollOf(SPL_FIREBALL)) {
```

---

Possibly also worth considering...

```suggestion
			if (spelldata[item._iSpell].sTargeted) {
```